gp/cartesian_graph.py

```diff
@@ -292,6 +295,22 @@ def to_sympy(self, simplify=True):
         List[sympy.core.Expr]
             List of sympy expressions.
         """
+        import sympy
```

Comment:
You seem to use two different strategies for `sympy` and `torch`: For `torch`, you try to import it and then define the function regardless of whether the import worked or not. For `sympy`, you import the package inside the function definition. Is either of these ways safe?

It seems that even if importing `torch` failed, the user can call the `to_torch()` method and receive an error? For `sympy`, the user can also call `to_sympy()` and receive an `ImportError` then?

I am wondering whether if might be better to set a global variable if the import failed and then replace the function definition in that case by a simple `NotImplementedError`? Something like:

```python
try:
    import torch  # noqa: F401
    TORCH_FOUND = True
except ModuleNotFoundError:
    TORCH_FOUND = False

 ...

def to_torch(self):
    if not TORCH_FOUND:
        raise NotImplementedError("No torch installation found. Method not supported. Consider installing the extra requirements via `pip install python-gp[torch]`")
    else:
         # Define function
```

Reply:
you are right, this was born out of not understanding python imports :( i would prefer to only import the modules where needed. for sympy this works, but for torch it creates problems with the dynamic creation of torch classes -- for some reason i don't understand.
i will change everything to a global import with an exception at the respective points where the module would be used. thanks for catching this! :+1: